From c47d66b5ca271a6fb7dd3ca9b9f4c8c0ac95f274 Mon Sep 17 00:00:00 2001
From: Samsuik <40902469+Samsuik@users.noreply.github.com>
Date: Mon, 10 Jul 2023 15:05:10 +0100
Subject: [PATCH] Limit firework spawn offset

Hacked clients can spawn fireworks anywhere they like within 8 blocks
from the interacted block.

diff --git a/src/main/java/net/minecraft/server/ItemFireworks.java b/src/main/java/net/minecraft/server/ItemFireworks.java
index e193a077b..f979704d5 100644
--- a/src/main/java/net/minecraft/server/ItemFireworks.java
+++ b/src/main/java/net/minecraft/server/ItemFireworks.java
@@ -6,6 +6,11 @@ public class ItemFireworks extends Item {
 
     public boolean interactWith(ItemStack itemstack, EntityHuman entityhuman, World world, BlockPosition blockposition, EnumDirection enumdirection, float f, float f1, float f2) {
         if (!world.isClientSide) {
+            // Blossom start - cap firework offset
+            f = MathHelper.a(f, 0.0f, 1.0f);
+            f1 = MathHelper.a(f1, 0.0f, 1.0f);
+            f2 = MathHelper.a(f2, 0.0f, 1.0f);
+            // Blossom end
             EntityFireworks entityfireworks = new EntityFireworks(world, (double) ((float) blockposition.getX() + f), (double) ((float) blockposition.getY() + f1), (double) ((float) blockposition.getZ() + f2), itemstack);
 
             world.addEntity(entityfireworks);
-- 
2.50.1

